src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -449,28 +450,40 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, jar_file, zinc_args,
       additional_snapshots +
       (self.extra_resources_digest(ctx),)
     )
-
-
-    argv = image_specific_argv + zinc_args
-    # TODO(#6071): Our ExecuteProcessRequest expects a specific string type for arguments,
-    # which py2 doesn't default to. This can be removed when we drop python 2.
-    argv = [text_type(arg) for arg in argv]
-
-    req = ExecuteProcessRequest(
-      argv=tuple(argv),
-      input_files=merged_input_digest,
-      output_files=(jar_file,) if self.get_options().use_classpath_jars else (),
-      output_directories=() if self.get_options().use_classpath_jars else (classes_dir,),
-      description="zinc compile for {}".format(ctx.target.address.spec),
-      jdk_home=self._zinc.underlying_dist.home,
-    )
-    res = self.context.execute_process_synchronously_or_raise(
-      req, self.name(), [WorkUnitLabel.COMPILER])
-
-    # TODO: Materialize as a batch in do_compile or somewhere
-    self.context._scheduler.materialize_directories((
-      DirectoryToMaterialize(get_buildroot(), res.output_directory_digest),
-    ))
+    BATCH_SIZE = 10
```

Comment:
This will need to be configurable, with a default value that does not trigger any batching (0?).